front-end/src/views/function/components/trigger.vue

```diff
@@ -0,0 +1,160 @@
+<template>
+  <el-drawer v-model="visible" :size="460" @close="onClose">
+    <el-form ref="TriggerForm" :model="TriggerForm" :rules="rules">
+      <el-row>
+        <el-form-item v-if="currentFunction.name" prop="functionName">
+          <span>functionName</span>
+          <el-select
+            v-model="TriggerForm.functionName"
+            placeholder="please select a function"
+            style="width: 368px"
+            size="small"
+          >
+            <el-option v-for="item in data" :key="item.index" :label="item.name" :value="item.key">
+              {{ item.name }}
+            </el-option>
+          </el-select>
+        </el-form-item>
+      </el-row>
+      <el-row>
+        <el-form-item prop="data">
+          <span>data</span>
+          <el-input
+            v-model="TriggerForm.data"
+            autosize
+            type="textarea"
+            placeholder="please enter the data"
+            style="width: 368px"
+          />
+        </el-form-item>
+      </el-row>
+    </el-form>
+    <el-row type="flex" justify="end">
+      <el-button type="primary" :loading="triggering" style="margin-bottom: 24px" @click="onSub('TriggerForm')">
+        Trigger
+      </el-button>
+    </el-row>
+    <el-card class="box-card">
+      <div class="clearfix">
+        <span>Result</span>
+        <span style="float: right; padding: 3px 0">{{ triggerResultType }}</span>
+      </div>
+      <div :style="{ minHeight: '64px' }" class="text item">
+        <span style="word-break: break-all">{{ triggerResult }}</span>
+      </div>
+    </el-card>
+  </el-drawer>
+</template>
+
+<script>
+  import { triggerFunc } from '@/api/func'
+
+  export default {
+    name: 'TriggerVue',
+    props: {
+      visible: {
+        type: Boolean,
+        default: false
+      },
+      visibleTrigger: {
+        type: Boolean,
+        default: false
+      },
+      data: {
+        type: Array,
+        default: () => []
+      },
+      currentFunction: {
+        type: Object,
+        default: () => {}
+      }
+    },
+    data() {
+      return {
+        TriggerForm: {},
+        triggerResult: '',
+        triggerResultType: '',
+        triggering: false,
+        rules: {
+          functionName: [{ required: true, message: 'Please select a function!', trigger: 'blur' }],
+          data: [{ required: true, message: 'Please select a function!', trigger: 'blur' }]
+        }
+      }
+    },
+    watch: {
+      currentFunc() {
+        this.TriggerForm.setFieldsValue({
+          functionName: this.currentFunction.name
+        })
+      }
+    },
+    methods: {
+      onClose () {
```

Comment:
Codacy has a fix for the issue: [Delete `·`](https://app.codacy.com/gh/FunctionStream/function-stream/pullRequest?prid=7834758)
```suggestion
      onClose() {
```